packages/firestore/test/integration/remote/stream.test.ts

```diff
@@ -262,4 +281,51 @@ describe('Write Stream', () => {
         });
     }, queue);
   });
+
+  it('force refreshes auth token on receiving unauthenticated error', () => {
+    const queue = new AsyncQueue();
+    const credentials = new MockCredentialsProvider();
+
+    return withTestDatastore(
+      ds => {
+        const writeStream = ds.newPersistentWriteStream();
+        writeStream.start(streamListener);
+        return streamListener
+          .awaitCallback('open')
+          .then(() => {
+            // Simulate callback from GRPC with an unauthenticated error -- this should invalidate
+            // the token.
+            writeStream.handleStreamClose(
+              new FirestoreError(Code.UNAUTHENTICATED, '')
+            );
+            return streamListener.awaitCallback('close');
+          })
+          .then(() => {
+            writeStream.start(streamListener);
+            return streamListener.awaitCallback('open');
+          })
+          .then(() => {
+            // Simulate a different error -- token should not be invalidated this time.
+            writeStream.handleStreamClose(
+              new FirestoreError(Code.UNAVAILABLE, '')
+            );
+            return streamListener.awaitCallback('close');
+          })
+          .then(() => {
+            writeStream.start(streamListener);
+            return streamListener.awaitCallback('open');
+          })
+          .then(() => {
+            expect(credentials.observedStates).to.have.ordered.members([
```

Comment:
In practice, the error message isn't particularly helpful (along the lines of `Array(4) does not equal Array(4)`. From a quick search, I only found references to old versions of Chai. Let me know if I'm using wrong comparators.